backend/internal/authn/authnhandler.go

```diff
@@ -20,129 +20,167 @@
-	logger.Info("Initializing authentication flow.")
-
-	// Handle the authentication request with the selected authenticator.
-	selectedAuthenticator := r.URL.Query().Get("authenticator")
-	authr := getAuthenticator(selectedAuthenticator)
-	logger.Debug("Selected authenticator: ", log.String("authenticator", authr.GetName()))
+	logger.Debug("Handling authentication request")
 
-	// Check if the session data is already stored with a session data key.
-	sessionDataKey := ctx.SessionDataKey
-	sessionDataStore := sessionstore.GetSessionDataStore()
-	ok, sessionData := sessionDataStore.GetSession(sessionDataKey)
-	if !ok {
-		http.Error(w, "Session data not found for session data key", http.StatusBadRequest)
+	authR, err := systemutils.DecodeJSONBody[model.AuthNRequest](r)
+	if err != nil {
+		writeAPIErrorResponse(w, logger, constants.APIErrorJSONDecodeError, http.StatusBadRequest)
 		return
 	}
 
-	sessionData.CurrentAuthenticator = authr.GetName()
+	// Sanitize the inputs
+	sessionDataKey := systemutils.SanitizeString(authR.SessionDataKey)
+	flowID := systemutils.SanitizeString(authR.FlowID)
+	actionID := systemutils.SanitizeString(authR.ActionID)
+	inputs := systemutils.SanitizeStringMap(authR.Inputs)
 
-	// Store session data in the session store. This replaces the old session data with the new one.
-	sessionDataStore.AddSession(ctx.SessionDataKey, sessionData)
-
-	err := authr.Process(w, r, ctx)
-	if err != nil {
-		utils.RedirectToErrorPage(w, r, oauthconstants.ErrorServerError,
-			"Failed to process authentication request")
+	if sessionDataKey == "" && flowID == "" {
+		writeAPIErrorResponse(w, logger, constants.APIErrorInvalidRequest, http.StatusBadRequest)
+		return
 	}
-}
-
-// HandleAuthenticationRequest handles the authentication request received.
-func (ah *AuthenticationHandler) HandleAuthenticationRequest(w http.ResponseWriter, r *http.Request) {
-	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, "AuthenticationHandler"))
-	logger.Info("Handling authentication request.")
-
-	var sessionDataKey, state string
-	if r.Method == http.MethodGet {
-		logger.Debug("Processing GET request received for authentication flow endpoint.")
 
-		// Extract session data key from the query parameters.
-		sessionDataKey = r.URL.Query().Get(oauthconstants.SessionDataKey)
-		state = r.URL.Query().Get(oauthconstants.State)
-	} else {
-		logger.Debug("Processing POST request received for authentication flow endpoint.")
+	var sessionData sessionmodel.SessionData
+	appID := ""
 
-		// Parse form data.
-		if err := r.ParseForm(); err != nil {
-			http.Error(w, "Failed to parse form data", http.StatusBadRequest)
+	// Check if the session data is already stored with a session data key.
+	sessionDataStore := sessionstore.GetSessionDataStore()
+	if sessionDataKey != "" {
+		var ok bool
+		ok, sessionData = getSessionData(sessionDataStore, sessionDataKey, w, logger)
```

Comment:
Lets make this `getAndRemoveSessionData()`?

Reply:
This helper method doesn't remove the session data and removal also happens conditionally after invoking this method

Reply:
When this method is invoked in line 119, the session is not removed when the status is incomplete